_posts/you-wanna-test-http.md

```diff
@@ -0,0 +1,215 @@
+---
+layout: post
+header_image: /img/posts/you-wanna-test-http/header.jpg
+title: "You wanna test HTTP, right?"
+author: Mauro Servienti
+synopsis: "Tests generate confidence. Awesome. Sometimes we need to test components requiring HTTP, and we risk building complex test infrastructures that only erode confidence. What options do we have?"
+tags:
+- unit test
+- integration test
+- HTTP
+---
+
+One of the beauties of testing is that it generates confidence. It's genuinely the essential aspect. Code evolves, needs to be maintained and fixed. We want to be confident that a set of changes will not change the expected behavior; we need to be sure we're not reintroducing bugs or undesirable behaviors that we already fixed.
+
+There is a need for safety, and testing creates that.
+
+Testing practices alone are not enough to create the required safety. Manual testing, for example, is a form of testing that comes with little sense of security: manual testing is prone to human and distraction errors. To build the required level of trust, we need attributes such as repeatability, isolation, and automation. Tests need to be automated so that they can run automatically at well-known predefined stages. Tests need to be repeatable and idempotent to guarantee consistency of results; Finally, tests need to be isolated and self-contained to ensure that one test doesn't affect others.
```

Comment:
```suggestion
But testing practices alone are not enough to create the required safety. Manual testing, for example, is a form of testing that comes with little sense of security: manual testing is prone to human and distraction errors. To build the required level of trust, we need attributes such as repeatability, isolation, and automation. Tests need to be automated so that they can run automatically at well-known predefined stages. Tests need to be repeatable and idempotent to guarantee consistency of results. Finally, tests need to be isolated and self-contained to ensure that one test doesn't affect others.
```